data_validation/cli_tools.py

```diff
@@ -436,316 +394,414 @@ def _configure_validate_parser(subparsers):
+        help="Comma separated tables list in the form 'schema.table=target_schema.target_table'",
+    )
+    _add_common_arguments(optional_arguments, required_arguments)
+
 
 def _configure_schema_parser(schema_parser):
     """Configure arguments to run schema level validations."""
-    _add_common_arguments(schema_parser)
-    schema_parser.add_argument(
+
+    # Group optional arguments
+    optional_arguments = schema_parser.add_argument_group("optional arguments")
+    optional_arguments.add_argument(
         "--exclusion-columns",
         "-ec",
         help="Comma separated list of columns 'col_a,col_b' to be excluded from the schema validation",
     )
-    schema_parser.add_argument(
+    optional_arguments.add_argument(
         "--allow-list",
         "-al",
         help="Comma separated list of datatype mappings due to incompatible datatypes in source and target. e.g: decimal(12,2):decimal(38,9),string[non-nullable]:string",
     )
 
+    # Group required arguments
+    required_arguments = schema_parser.add_argument_group("required arguments")
+    required_arguments.add_argument(
+        "--tables-list",
+        "-tbls",
+        default=None,
+        required=True,
+        help="Comma separated tables list in the form 'schema.table=target_schema.target_table'",
+    )
+    _add_common_arguments(optional_arguments, required_arguments)
+
 
 def _configure_custom_query_parser(custom_query_parser):
     """Configure arguments to run custom-query validations."""
-    _add_common_arguments(custom_query_parser)
-    custom_query_parser.add_argument(
-        "--custom-query-type",
-        "-cqt",
-        required=True,
-        choices=["row", "column"],
-        help="Which type of custom query (row/column)",
+
+    custom_query_subparsers = custom_query_parser.add_subparsers(
+        dest="custom_query_type"
+    )
+
+    # Add arguments for custom-query row parser
+    custom_query_row_parser = custom_query_subparsers.add_parser(
+        "row", help="Run a custom query row validation"
+    )
+    _configure_custom_query_row_parser(custom_query_row_parser)
+
+    # Add arguments for custom-query column parser
+    custom_query_column_parser = custom_query_subparsers.add_parser(
+        "column", help="Run a custom query column validation"
     )
-    custom_query_parser.add_argument(
+    _configure_custom_query_column_parser(custom_query_column_parser)
+
+
+def _configure_custom_query_row_parser(custom_query_row_parser):
+    # Group optional arguments
+    optional_arguments = custom_query_row_parser.add_argument_group(
+        "optional arguments"
+    )
+    optional_arguments.add_argument(
+        "--filters",
+        "-filters",
+        help="Filters in the format source_filter:target_filter",
+    )
+    optional_arguments.add_argument(
+        "--threshold",
+        "-th",
+        type=threshold_float,
+        help="Float max threshold for percent difference",
+    )
+    optional_arguments.add_argument(
```

Comment:
We don't support random-row for custom query

Reply:
Done. Removed `random-row` for `custom-query`